docs/advanced/tls_hardware_acceleration.md

```diff
@@ -0,0 +1,131 @@
+# mbed TLS Hardware Acceleration
+
+This document explains how to use hardware acceleration for various cryptographic primitives with [mbed TLS](https://github.com/ARMmbed/mbedtls) running on an embedded device.
+
+
+## Introduction
+
+### Why should I add hardware acceleration?
+
+Whether the application developer uses mbed TLS as a cryptographic library or uses it as a TLS stack, they will end up with time-consuming cryptographic routines in their application that will heavily impact response time. The platform that reduces this adverse effect will have an edge over platforms that do not (or do so to a lesser extent).
```

Comment:
I am happy to change it, I just would like to make clear, that I didn't mention time-constantness, the word was "time-consuming".